tools/codesigningtool/codesigningtool.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2018 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import base64
+import json
+import os
+import plistlib
+import re
+import subprocess
+import sys
+
+
+def _check_output(args, inputstr=None):
+    proc = subprocess.Popen(args,
+        stdin=subprocess.PIPE,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE)
+    return proc.communicate(input=inputstr)[0]
+
+
+def _parse_mobileprovision_file(mobileprovision_file):
+    """Reads and parses a mobileprovision file."""
+    plist_xml = subprocess.check_output([
+        "security", "cms",
+        "-D",
+        "-i", mobileprovision_file,
+    ])
+    if hasattr(plistlib, "readPlistFromString"):
```

Comment:
As in #271, lets just try/except to pick between the methods so we're consistent.